dev/NavigationView/NavigationView.cpp

```diff
@@ -1273,10 +1285,16 @@ void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const
         if (isSelectionSuppressed)
         {
             UndoSelectionAndRevertSelectionTo(prevItem, nextActualItem);
-
+            
             // Undo only happened when customer clicked a selectionsuppressed item. 
             // To simplify the logic, OnItemClick didn't raise the event and it's been delayed to here.
             RaiseItemInvoked(nextActualItem, isSettingsItem);
+
+            auto container = NavigationViewItemOrSettingsContentFromData(nextActualItem);
+            if (container)
+            {
+                ToggleIsExpanded(container);
+            }
```

Comment:
extract this as a function since we use it many times. for example: ToggleIsExpandedFromItemData